src/guidellm/core/result.py

```diff
@@ -150,19 +184,28 @@ class TextGenerationBenchmark(Serializable):
     This is a set of results and errors for a specific mode and rate.
     """
 
-    mode: str
-    rate: Optional[float]
-    results: List[TextGenerationResult] = []
-    errors: List[TextGenerationError] = []
-    concurrencies: List[RequestConcurrencyMeasurement] = []
+    mode: str = Field(description="The generation mode, either 'async' or 'sync'.")
+    rate: float = Field(
+        default=None, description="The requested rate of requests per second."
+    )
+    results: List[TextGenerationResult] = Field(
+        default_factory=list, description="The results of the text generation requests."
+    )
+    errors: List[TextGenerationError] = Field(
+        default_factory=list, description="The errors of the text generation requests."
+    )
+    concurrencies: List[RequestConcurrencyMeasurement] = Field(
+        default_factory=list,
+        description="The concurrency measurements of the requests.",
+    )
```

Comment:
Should we make `mode` an `Enum`?

Reply:
Yes, I don't see any issue with that. Let's move that to another enhancement PR, though